metals/src/main/scala/scala/meta/internal/metals/OnTypeFormattingProvider.scala

```diff
@@ -0,0 +1,71 @@
+package scala.meta.internal.metals
+
+import org.eclipse.lsp4j.{DocumentOnTypeFormattingParams, Range, TextEdit}
+
+import scala.meta.internal.mtags.Semanticdbs
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.meta.inputs.Input
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.tokens.Token
+import scala.meta.tokens.Token.Constant
+
+final class OnTypeFormattingProvider(
+    semanticdbs: Semanticdbs
+)(implicit ec: ExecutionContext) {
+
+  val quot = """\u0022\u0022\u0022"""
+  val space = " "
+
+  private def indent(toInput: Input.VirtualFile, pos: meta.Position): String = {
+    val bit = toInput.text.substring(0, pos.start)
+    val lastNewline = bit.lastIndexOf("\n")
+    val lastPipe = bit.lastIndexOf("|")
+    val length = toInput.text
+      .substring(lastNewline, lastPipe)
+      .length - 1
+    space * length
+  }
+
+  private def longString(text: String, token: Token): Boolean = {
```

Comment:
`isLongString` sound a bit nicer nicer when read inside an if: 
`if((...) and isLongString(foo)`